provider/image.go

```diff
@@ -90,8 +90,17 @@ func (p *dockerNativeProvider) dockerBuild(ctx context.Context,
 		return "", nil, err
 	}
 
-	// make the build context
-	tar, err := archive.TarWithOptions(img.Build.Context, &archive.TarOptions{})
+	// make the build context and ensure to exclude dockerignore file patterns
+	dockerIgnorePath := filepath.Join(build.Context, ".dockerignore")
+	ignorePatterns, err := getIgnore(dockerIgnorePath)
+	if err != nil {
+		return "", nil, err
+	}
+
+	tar, err := archive.TarWithOptions(img.Build.Context, &archive.TarOptions{
+		ExcludePatterns: ignorePatterns,
+		IncludeFiles:    []string{"Dockerfile", ".dockerignore"},
```

Comment:
Do we need to change "Dockerfile" to match `img.Build.Dockerfile`?

Reply:
Examples like this one likely need it:

https://github.com/pulumi/pulumi-docker/blob/6ee37787d94d210effb3232dcdd0a2696f8fb69f/examples/aws/index.ts#L49-L57